execution/types/unchecked_extrinsic.go

```diff
@@ -20,43 +20,46 @@ const (
 	ExtrinsicUnmaskVersion = 0b0111_1111
 )
 
+type PayloadInitializer = func(call primitives.Call, extra primitives.SignedExtra) (
+	primitives.SignedPayload, primitives.TransactionValidityError,
+)
+
 type uncheckedExtrinsic struct {
 	version sc.U8
 	// The signature, address, number of extrinsics have come before from
 	// the same signer and an era describing the longevity of this transaction,
 	// if this is a signed extrinsic.
-	signature sc.Option[primitives.ExtrinsicSignature]
-	function  primitives.Call
-	extra     primitives.SignedExtra
-	crypto    io.Crypto
+	signature          sc.Option[primitives.ExtrinsicSignature]
+	function           primitives.Call
+	extra              primitives.SignedExtra
+	payloadInitializer PayloadInitializer
+	crypto             io.Crypto
+	hashing            io.Hashing
 }
 
 // NewUncheckedExtrinsic returns a new instance of an unchecked extrinsic.
-func NewUncheckedExtrinsic(
-	version sc.U8,
-	signature sc.Option[primitives.ExtrinsicSignature],
-	function primitives.Call,
-	extra primitives.SignedExtra,
-) uncheckedExtrinsic {
-
+func NewUncheckedExtrinsic(version sc.U8, signature sc.Option[primitives.ExtrinsicSignature], function primitives.Call, extra primitives.SignedExtra) uncheckedExtrinsic {
 	return uncheckedExtrinsic{
-		version:   version,
-		signature: signature,
-		function:  function,
-		extra:     extra,
-		crypto:    io.NewCrypto(),
+		version:            version,
+		signature:          signature,
+		function:           function,
+		extra:              extra,
+		payloadInitializer: primitives.NewSignedPayload,
+		crypto:             io.NewCrypto(),
+		hashing:            io.NewHashing(),
 	}
 }
 
 // NewUnsignedUncheckedExtrinsic returns a new instance of an unsigned extrinsic.
-func NewUnsignedUncheckedExtrinsic(function primitives.Call) primitives.UncheckedExtrinsic {
+func NewUnsignedUncheckedExtrinsic(function primitives.Call) uncheckedExtrinsic {
```

Comment:
This should return the interface, it's odd a public method to return a private struct